src/main/java/com/github/scr/j8iterables/core/CloseableSpliterator.java

```diff
@@ -0,0 +1,72 @@
+package com.github.scr.j8iterables.core;
+
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.util.Comparator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+
+/**
+ * A {@link Spliterator} that closes a {@link Closeable} resource when it is completed.
+ *
+ * @author scr on 2/24/17.
+ * @apiNote Upon close, any {@link java.io.IOException} is converted to {@link java.io.UncheckedIOException}.
+ */
+public class CloseableSpliterator<T, R> implements Spliterator<T> {
```

Comment:
You can see from your observation of initial javadoc being bogus now, that I went with the `AutoCloseable` approach, but I abandoned it as cumbersome because of that darned `throws Exception` clause :(

By using a method, it forces the caller to decide and therefor know that, for instance, `IOException` will be converted to `UncheckedIOException`.